### PR TITLE
Patches pagination in Elementor product widget.

### PR DESCRIPTION
**Pull request by Mauricio-Urrego**
### Ticket
- [Elementor: pagination clicks reload full page - no anchor](https://app.asana.com/0/30156186242982/1200264058586621/f)

### Description
- Relevant issue: https://github.com/elementor/elementor/issues/4703

Patched by adding an id tag to the widget that page links can jump to.

### CODE REVIEW

File: assets/scripts/main.js [hunk 1]
```diff
@@ -110,4 +110,14 @@
       $('.wcppec-checkout-buttons').hide();
     }
   });
+  
+  // Fixes missing anchor on elementor widget pagination.
+  // Must be combined with <a name="widget"></a> in title.
+  // https://github.com/elementor/elementor/issues/4703
+  $(document).ready(() => {
+    $('.elementor-widget-woocommerce-products').attr('id', $('.elementor-widget-woocommerce-products').attr('data-id'));
```
Comment (sun) [inline assets/scripts/main.js:118]:
```suggestion
    const $widget = $('.elementor-widget-woocommerce-products');
    if (!$widget.attr('id')) {
      $widget.attr('id', $widget.attr('data-id'));
    }
```
You can use the same `$widget` as context for the selector looking for all links below.

File: assets/scripts/main.js [hunk 2]
```diff
@@ -110,4 +110,14 @@
...
+    $('.elementor-widget-woocommerce-products .page-numbers a').each((i, a) => {
+      $(a).attr('href', $(a).attr('href') + '#' + $('.elementor-widget-woocommerce-products').attr('id'));
```
Comment (sun) [inline assets/scripts/main.js:120]: The first argument (which you named `i`) should contain the full jQuery object of the link already, no?

In addition, there may be multiple product listing widgets on a single page, so you should find the closest to the link:
```suggestion
      $(a).attr('href', $(a).attr('href') + '#' + $(a).closest('.elementor-widget-woocommerce-products').attr('id'));
```

**Review by sun — CHANGES_REQUESTED** [review]

**Comment by sun** [comment]
@Mauricio-Urrego I tested and my last suggestions on this PR should work correctly. Please have another look.

image

**Review by sun — APPROVED** [review]

**Comment by sun** [comment]
The code should work, seems like something else is conflicting.

**Review by fabianmarz — CHANGES_REQUESTED** [review]
Blocking for now as the compiled file is missing.

**Comment by fabianmarz** [comment]
Just a quick heads up. This is not working because some other elementor function (`handleAnchorLinks`)[https://github.com/elementor/elementor/blob/2cd3d50d8d87edde9cd3c820f5a0f8b1f2c4f24f/assets/dev/js/frontend/utils/anchors.js#L64] code is hooking into all anchors and puts a `e.preventDefault` if it's an internal link 🤔 

I'm currently looking into a viable solution.

**Comment by Mauricio-Urrego** [comment]
> Just a quick heads up. This is not working because some other elementor function (`handleAnchorLinks`)[https://github.com/elementor/elementor/blob/2cd3d50d8d87edde9cd3c820f5a0f8b1f2c4f24f/assets/dev/js/frontend/utils/anchors.js#L64] code is hooking into all anchors and puts a `e.preventDefault` if it's an internal link 🤔
> 
> @Mauricio-Urrego, could you please test my commit? I remove the click event listeners to prevent elementor from treating these links as anchor target links. I guess it's save to remove the click listeners all together as there should not be any attached.

Works for me! Just need to commit compiled assets then?

**Review by fabianmarz — APPROVED** [review]

**Pull request merged**